### PR TITLE
Jwt authentication backend

### PR DESCRIPTION
**Pull request by Pulkit95700**
### Issue Title: JWT Authenticator with Role based Authorization #122

- **Name of the project** : JWT Authenticatior With Roles
- **Category:** NodeJs, API, ExpressJs, MongoDB
- **Name:** Pulkit Gupta
- **GitHub ID:** Pulkit95700
- **Idenitfy yourself:** GSSOC
- **Fix:** Fixes Avdhesh-Varshney/WebMasterLog#122

<br class="Apple-interchange-newline">
</body>
</html>

### Describe the add-ons or changes you've made 📃
Created full Project for JWT authentication

Directory Structure
- images 
   - contains the images for the Working.md file in base directory of the project
- src
   - index.js
   - constants.js
   - app.js
   - utils
       - ApiError.js
       - ApiResponse.js
       - asyncHandler.js
   - controllers
       - user.controller.js
   - db 
       - index.js
   - middlewares
       - auth.middleware.js
       - permit.middleware.js
   - models
       - user.model.js
   - routes
       - user.routes.js
   - .env
   - package.json
   - package-lock.json
   - Working.md
  

### How Has This Been Tested? ⚙️

This API has undergone extensive testing using Postman, covering secure endpoints and ensuring all functions are bug-free. We've also conducted end-to-end logic testing to ensure smooth operation.
### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CONTRIBUTING.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Note to reviewers 📄

This api helps user to get a modern starting project for production use. This is a standard setup for any production level project. The different files are used to handle different functions. You can clone it to get started on your project quickly.

### CODE REVIEW

**Comment by Pulkit95700** [comment]
@Avdhesh-Varshney Please check pr for the following issue.
Fix: Fixes Avdhesh-Varshney/WebMasterLog#122

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@Pulkit95700 Great Work.
But i require 1 screenshot out of these images. So, out of these images, convert 1 image into webp format and keeps outside the images directory.
And don't forget to update the README file OK.

**Comment by Pulkit95700** [comment]
Hello @Avdhesh-Varshney, I have added an image in webp format. Also updated the README file in the Intermediate directory. You can check the changes.

File: Node-JS-Projects/Intermediate/JWT-Authenticator-With-Roles/image.webp
Comment (Avdhesh-Varshney) [inline Node-JS-Projects/Intermediate/JWT-Authenticator-With-Roles/image.webp:1]: Yah, it's good, but name of ss should be `screenshot.webp`.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@Pulkit95700 For further communication join discord server [link](https://discord.gg/tSqtvHUJzE).

File: Node-JS-Projects/Intermediate/JWT-Authenticator-With-Roles/image.webp
Comment (Pulkit95700) [reply Node-JS-Projects/Intermediate/JWT-Authenticator-With-Roles/image.webp:1]: did it.

**Review by Avdhesh-Varshney — APPROVED** [review]
@Pulkit95700 PR Approved 🎉

**Pull request merged**